mrmustard/physics/__init__.py

```diff
@@ -87,7 +88,7 @@ def overlap(A, B) -> float:
     Returns:
         float: the overlap between the two states
     """
-    raise NotImplementedError
```

Comment:
How come an exception was raised here? Was this just a mistake/bug? 🤔

Reply:
I believe this functions were either not correctly implemented or thoroughly tested before release so they were set as not implemented.

I'm leaving them untouched for this PR and will check implementation+tests on a dedicated pull request.